themes/default/content/learn/abstraction-encapsulation/component-resources/index.md

```diff
@@ -29,18 +29,91 @@ A resource in Pulumi terms is basically a building block. That building block co
 
 Why would you make a component resource instead of just a "simple" logical grouping that you can call and use like a generic library full of classes and functions? A component resource shows up in the Pulumi ecosystem just like any other resource. That means it has a trackable state, appears in diffs, and has a name field you can reference that refers to the entire grouping.
 
-We've actually already started creating a component resource in the encapsulation part when we made a new class that built up an `s3.Bucket` and an `s3.BucketPolicy` with a `define_policy` function. Let's now turn that logical grouping into an actual component resource.
+We've actually already started creating a component resource in the encapsulation part when we made a new class that built up an `s3.Bucket` and an `s3.BucketPolicy`. Let's now turn that logical grouping into an actual component resource.
```

Comment:
I only did this to avoid having to duplicate the paragraph, since the TypeScript example doesn't use this additional step. (It just defines the policy in the constructor.)

Reply:
Gotcha; makes sense.